compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/configschema/TypeConverter.java

```diff
@@ -173,14 +182,13 @@ private JsonObject generateRecordType(BRecordType effectiveType) {
             typeNode.add("required", requiredFields);
         }
         // Get record type and set the type name as a property
-        if (effectiveType.getIntersectionType().isPresent()) {
-            for (BType bType : effectiveType.getIntersectionType().get().getConstituentTypes()) {
-                // Does not consider anonymous records
-                if (bType instanceof BTypeReferenceType) {
-                    typeNode.addProperty("name", bType.toString().trim());
-                }
+        for (BType bType : intersectionType.getConstituentTypes()) {
+            // Does not consider anonymous records
+            if (bType instanceof BTypeReferenceType) {
```

Comment:
Can't we check `TYPE_REFERENCED_TYPE_TAG` here?

Reply:
Updated